climada/util/value_representation.py

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Mon Nov 16 19:21:42 2020
+
+@author: ckropf
+"""
+
+import math
+import numpy as np
+import logging
+
+LOGGER = logging.getLogger(__name__)
+
+ABBREV = {1:'',
+          1000: 'K',
+          1000000: 'M',
+          1000000000: 'Bn', 
+          1000000000000: 'Tn'}
+
+def sig_dig(x, n_sig_dig = 16):
+    """
+    Rounds x to n_sig_dig number of significant digits. 
+    Examples: 1.234567 -> 1.2346, 123456.89 -> 123460.0
+
+    Parameters
+    ----------
+    x : float
+        number to be rounded
+    n_sig_dig : int, optional
+        Number of significant digits. The default is 16.
+
+    Returns
+    -------
+    float
+        Rounded number
+
+    """
+    num_of_digits = len(str(x).replace(".", ""))
+    if n_sig_dig >= num_of_digits:
+        return x
+    n = math.floor(math.log10(abs(x)) + 1 - n_sig_dig)
+    result = np.round(x * 10**(-n)) * 10**n
+    return result
+
+def sig_dig_list(iterable, n_sig_dig=16):
+    """
+    Vectorized form of sig_dig. Rounds a list of float to a number
+    of significant digits
+
+    Parameters
+    ----------
+    iterable : iter(float)
+        iterable of numbers to be rounded
+    n_sig_dig : int, optional
+        Number of significant digits. The default is 16.
+
+
+    Returns
+    -------
+    list
+        list of rounded floats
+
+    """
+    return np.vectorize(sig_dig)(iterable, n_sig_dig)
+
+
+def value_to_monetary_unit(values, n_sig_dig=None, abbreviations=None):
+    """
+    Converts values to closest common monetary unit, default: (K, M Bn, Tn)
+
+    Parameters
+    ----------
+    values : float, list(float) or np.ndarray
+        Values to be converted
+    n_sig_dig : int, optional
+        Number of significant digits to return.
+        Examples n_sig_di=5: 1.234567 -> 1.2346, 123456.89 -> 123460.0
+        Default: all digits are returned. 
+    abbreviations: dict, optional
+        Name of the abbreviations for the money 1000s counts
+        Default:  
+         {0:'',
+          1000: 'K',
+          1000000: 'M',
+          1000000000: 'Bn', 
+          1000000000000: 'Tn'}
+
+    Returns
+    -------
+    mon_val : np.ndarray
+        Array of values in monetary unit 
+    name : string
+        Monetary unit
+
+    """
+    
+    if not isinstance(values, list):
+        values = [values]
```

Comment:
The docstring says `ndarray` is a possible type too. That seems contradicting to just checking for a list.
In general, if multiple argument types are allowed, addressing them all in a `if isinstance(...)` block is probably a good idea. (Or none when duck-typing.)